lib/security/JwsToken.ts

```diff
@@ -68,8 +78,11 @@ export default class JwsToken extends JoseToken {
     const encodedContent = Base64Url.encode(this.content);
     // 3. Compute the headers
     const headers = jwsHeaderParameters || {};
-    headers['alg'] = jwk.defaultSignAlgorithm;
-    if (jwk.kid) {
+    // add required fields if required
+    if (!('alg' in headers)) {
+      headers['alg'] = jwk.defaultSignAlgorithm;
+    }
+    if (jwk.kid && !('kid' in headers)) {
```

Comment:
Sanity: If `jwk.kid` doesn't exist and `kid` is not in header, then we let it go even if it's required?